### PR TITLE
edk2_pr_eval: Add Policy 5

### PR DESCRIPTION
**Pull request by Javagedes**
Policy 5 is used when a library INF has been changed. Policy 5 will parse each package's dsc (Found via `<pkg>.ci.yaml`.`PrEval`.`DscPath`) and mark a package to be built if that package has a dependency on the changed library INF.

Consuming repo's must update each package's ci.yaml file to now include a new top-level section "PrEval" with an entry "DscPath". This will be the package's DSC used to compile all components and libraries the package provides.

``` yaml
{
    "PrEval" : {
        "DscPath": "path/to/dsc"
    }
}
```

### CODE REVIEW

File: edk2toolext/invocables/edk2_pr_eval.py [hunk 1]
```diff
@@ -308,6 +308,26 @@ def get_packages_to_build(self, possible_packages: list) -> dict:
                         remaining_packages.remove(p)  # remove from remaining packages
                         break
 
+        #
+        # Policy 5: If a file changed is a Library INF file, then build all packages that depend on that Library
+        #
+        for f in filter(lambda f: Path(f).suffix == ".inf", files):
+            for p in remaining_packages[:]:
+                dsc = Path(self.edk2_path_obj.GetAbsolutePathOnThisSystemFromEdk2RelativePath(p)) / f'{p}.dsc'
```
[the remark this answers is not in the document]
Comment (Javagedes) [reply edk2toolext/invocables/edk2_pr_eval.py:316]: Due to the fact that edk2_pr_eval is used to determine what packages need to be built using the compiler plugin for stuart_ci_build, I have changed this logic to find the package dsc path from the compiler plugin ci settings rather then assuming it is <package>/<package>.dsc

File: edk2toolext/invocables/edk2_pr_eval.py [hunk 2]
```diff
@@ -308,6 +309,27 @@ def get_packages_to_build(self, possible_packages: list) -> dict:
...
+        for f in filter(lambda f: Path(f).suffix == ".inf", files):
+            for p in remaining_packages[:]:
+                dsc = self._get_pkg_dsc_path(p)
+                if not dsc:
+                    logging.debug(f"Policy 5 - Package {p} skipped due to missing ci.dsc file or missing Compiler "
+                                  "Plugin DSC path.")
+                    continue
+
+                dsc_parser = DscParser()
+                dsc_parser.SetBaseAbsPath(self.edk2_path_obj.WorkspacePath)
```
Comment (Javagedes) [inline edk2toolext/invocables/edk2_pr_eval.py:325]: Use `SetEdk2Path()`

**Comment by Javagedes** [comment]
Per external conversation, we will be adding a new section to a package's ci.yaml file for preval. The proposed section is as follows:

<Pkg>.ci.yml
``` yaml
{
    "PrEval": {
        "DscPath": "<Pkg/relative/path/to/dsc>"
    }
}
```

PrEval will also grab the "Defines" section from a ci.yaml file and add that to the environment when parsing the dsc.

**Comment by Javagedes** [comment]
Integration tests will fail due to missing PrEval section in package's CI file.

**Comment by makubacki** [comment]
Can you please clarify in the PR description what platforms need to do? Clearly state what they need to change for what purpose and what will not work if they don't.

**Review by makubacki — APPROVED** [review]

**Comment by Javagedes** [comment]
> Can you please clarify in the PR description what platforms need to do? Clearly state what they need to change for what purpose and what will not work if they don't.

Absolutely. I'll also add it as an integration instruction in the release.

**Pull request merged**